test/Microsoft.AspNetCore.Server.KestrelTests/TestHelpers/MockSocket.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Threading;
+using Microsoft.AspNetCore.Server.Kestrel.Infrastructure;
+using Microsoft.AspNetCore.Server.Kestrel.Networking;
+
+namespace Microsoft.AspNetCore.Server.KestrelTests.TestHelpers
+{
+    class MockSocket : UvStreamHandle
+    {
+        public MockSocket(int threadId, IKestrelTrace logger) : base(logger)
+        {
+            var uv = new MockLibuv();
+            CreateMemory(uv, threadId, uv.handle_size(Libuv.HandleType.STREAM));
```

Comment:
I'll need to check again - at some point I got an exception about it being invalid. But that was when I was playing around with the code, maybe the way it is now it doesn't matter anymore.